dradis/dradis-ce
# Notes tab in node view

### Spec
Currently, there is no way of deleting multiple notes in a node in one go. 

**Proposed solution**
Adding an evidence tab in Nodes#show that has a table with checkboxes for each record should solve the problem

### How to test
1. Add a node.
2. Add multiple notes to the node (about 20)
Useful code: 
```
30.times {|i| Note.create!(text: "Note #{i}", node_id: 27, category: Category.default)}
```
(27 was my node_id, use yours)

3. Assert that the tab works as expected:
 - assert that when the mouse is over a note row, edit/delete links appear for that row. Check that those links work
  - assert that when a note is selected, the "Delete" button shows up on the toolbar above
  - assert that by clicking the select all checkbox, all notes can be selected unselected. Check also that this works clicking not only in the checkbox, but also on its surrounding grey box
  - check that notes can be filtered with the search box in the top right
  - delete a few notes (< 15) using the checkboxes and the delete button in the toolbar. The notes should be deleted inline, and a "Notes deleted" message should appera. Notes should be removed from the table and the sidebar
   - delete a few notes (> 15) usning the checkboxes and the delete button in the sidebar. A console should popup and notes should be delete in the background. When the process finished we should see a 'Worker process completed' log message, and a button to close the console must appear. When closing the button the page should reload (No deleted notes on the sidebar and on the table)
